dqiv_patch.py

```
@@ -500,5 +514,202 @@ def patch_file_en(filename):
+    "us" + "/data/data/MESS/en"
+    for f in os.listdir(path_to_en_ds_files):
+        shutil.copy(path_to_en_ds_files + "/" + f, "en/" + f)
+
+    #list of mpts to extract from the obb
+    mpt_list = ['assets/msg/en/b0500000.mpt', 'assets/msg/en/b0501000.mpt', 'assets/msg/en/b0502000.mpt', 'assets/msg/en/b0503000.mpt', 'assets/msg/en/b0504000.mpt', 'assets/msg/en/b0505000.mpt', 'assets/msg/en/b0506000.mpt', 'assets/msg/en/b0507000.mpt', 'assets/msg/en/b0508000.mpt', 'assets/msg/en/b0509000.mpt', 'assets/msg/en/b0512000.mpt', 'assets/msg/en/b0513000.mpt', 'assets/msg/en/b0516000.mpt', 'assets/msg/en/b0517000.mpt', 'assets/msg/en/b0520000.mpt', 'assets/msg/en/b0521000.mpt', 'assets/msg/en/b0522000.mpt', 'assets/msg/en/b0523000.mpt', 'assets/msg/en/b0524000.mpt', 'assets/msg/en/b0525000.mpt', 'assets/msg/en/b0526000.mpt', 'assets/msg/en/b0527000.mpt', 'assets/msg/en/b0528000.mpt',
+            'assets/msg/en/b0529000.mpt', 'assets/msg/en/b0530000.mpt', 'assets/msg/en/b0531000.mpt', 'assets/msg/en/b0532000.mpt', 'assets/msg/en/b0533000.mpt', 'assets/msg/en/b0534000.mpt', 'assets/msg/en/b0535000.mpt', 'assets/msg/en/b0536000.mpt', 'assets/msg/en/b0537000.mpt', 'assets/msg/en/b0538000.mpt', 'assets/msg/en/b0539000.mpt', 'assets/msg/en/b0540000.mpt', 'assets/msg/en/b0541000.mpt', 'assets/msg/en/b0542000.mpt', 'assets/msg/en/b0543000.mpt', 'assets/msg/en/b0544000.mpt', 'assets/msg/en/b0545000.mpt', 'assets/msg/en/b0547000.mpt', 'assets/msg/en/b0548000.mpt', 'assets/msg/en/b0549000.mpt', 'assets/msg/en/b0550000.mpt', 'assets/msg/en/b0551000.mpt', 'assets/msg/en/b0552000.mpt']
+    
+    #extract the files and move the extracted files to root of en folder
+    print("Extracting files from obb...")
+    for mpt in mpt_list:
+        with ZipFile(path_to_roms + "/" + obb, 'r') as zObject:
+            zObject.extract(mpt, path="en/")
+            try:
+                os.rename("en/" + mpt, "en/" + mpt.split("assets/msg/en/")[1])
+            except:
+                pass
```

Not sure if it is a problem if this rename doesn't happen? If so would rather raise the exception so user is aware of it.

---

right. this was to fix an issue with windows. on linux when you rename a file and the target path for that file already exists, the file that was already there is overwritten, but on windows it raises an error.

I think this `try` statement can be completely removed due to the fact that extracting the rom when the mpt files are already in `en` is no longer something that happens.